using-timescaledb/adaptive-chunking.md

```diff
@@ -0,0 +1,130 @@
+mechanism within TimescaleDB automatically detects any "collisions"
+between new and old chunks due to the change in partitioning, and cuts
+chunks to fit within existing constraints.
+
+*Adaptive chunking* leverages the ability to set new
+`chunk_time_interval`s through automation. With adaptive chunking,
+however, the user instead sets a `chunk_target_size`, e.g., `1GB` and
+the system will try to adapt the `chunk_time_interval` on new chunks
+to reach this target chunk size using an algorithm. The system tries
+to adapt the first *open* (time) dimension, which is the same
+dimension that `chunk_time_interval` works on. An open dimension is
+one that *isn't* pre-partitioned, unlike hash-partitioned dimensions
+where the entire key space is divided into equal-size partitions.
+
+>:TIP: Setting a reasonable initial `chunk_time_interval` is still
+important because it allows the adaptive algorithm to more quickly
+reach the target chunk size. Setting a too large `chunk_time_interval`
+is worse than setting a too small one, because the system will be
+committed to that larger interval until the next chunk.  If no
+`chunk_time_interval` is set with adaptive chunking, the default is 1
+day.
+
+The default algorithm to calculate the next chunk's interval takes
+into account the *fill factor* of previous chunks, i.e., if they were
+over-filled or under-filled, and changes the interval
+accordingly. Note that the algorithm will never hit the exact target,
+and it will only change the `chunk_time_interval` if the chunk size
+deviates more than a certain threshold from the target size. The
+default algorithm should work well for workloads that see an increased
+data rate over time. It is also possible to implement [custom adaptive
+chunking algorithms](#custom-functions).
+
+>:TIP: It is recommended that hypertables with adaptive chunking
+enabled have an index on the dimension being adapted, typically the
+"time" dimension. Such an index is created by default, unless
+otherwise specified. The index is used by the default adaptive
+chunking algorithm to get the `min` and `max` value from a chunk to
+calculate its fill factor. Without an index, it must resort to a heap
+scan (i.e., scanning all the rows in a chunk), which is expensive. The
+system will print `WARNING`s in the log if has to do heap scans.
+
+### Usage
+
+To create a hypertable that uses adaptive chunking, simply call
+[`create_hypertable`][create-hypertable] as follows:
+
+```sql
+SELECT create_hypertable('conditions', 'time', chunk_target_size => '2GB');
+```
+
+This turns the table `conditions` into a hypertable that will try to
+adapt the time dimension to reach the given target chunk size.
+
+It is also possible to let the system *estimate* a target chunk size
+in case it is unclear what a good chunk size would be for the given
+system. In that case run:
+
+```sql
+SELECT create_hypertable('conditions', 'time', chunk_target_size => 'estimate');
```

Comment:
`SELECT chunk_target_size FROM _timescaledb_catalog.hypertable;` would work. But I agree that a getter might be useful. Still, we have no getters for other things, like `chunk_time_interval`. Maybe this can be a story about better/easier visibility into system settings in general?